### ISSUE TITLE
Parse Files saved hardcoded in json in array.

### ISSUE THREAD

**Issue by uzaysan** [parse-community/parse-server#7123]
### New Issue Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [ ] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [ ] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [ ] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description
I have files in json. And I put that json inside of an array. But seems like parse files are saved hardcoded. Even we change fileAdapter, or baseUrl in file adapter, files return with the old url.

### Steps to reproduce
Here is an example. This is How I save my files.

```
    for(var i = 0; i < mediaList.length; i++){
      //...other logic
    
           
      var mainmedia = new Parse.File(user.id, { base64: mediaData },mimeInfo.mime);
      var thumbfile = new Parse.File(user.id, { base64: thumbnail },mimeInfo.mime);
      

      var item = {"type":"image","width":ratiow,"height":ratioh,"media":mainmedia,"thumbnail":thumbfile};
      itemList.push(item);
    
    }
    post.set("media",itemList);
```

And  baseUrl in my adapter is : `http://files.example.com`

Now file is saved with this url: `http://files.example.com/filename`

But If  change baseUrl to this: `http://my-new-file-endpoint.example.com` 
I still get `http://files.example.com/filename` but I should get `http://my-new-file-endpoint.example.com/filename` 

### Actual Outcome
I get old url
 `http://files.example.com/filename`

### Expected Outcome
I should get  `http://my-new-file-endpoint.example.com/filename` 

### Failing Test Case / Pull Request

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment

Server
- Parse Server version: `4.5` //But got the same result in previous versions too
- Operating system: `Ubuntu 18.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Remote server`

Database
- System (MongoDB or Postgres): `Mongodb`
- Database version: `4.4`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `remote`

**Comment by mtrezza** [comment]
If I understand correctly from your test case, you are saving a Parse File with a full URL to the database. A Parse File normally does not contain the full URL but only the filename. The file adapter then adds the base URL part when retrieving the file. Hence, the URL you are currently saving to the DB does not change bu changing the `baseUrl` parameter.

**Comment by uzaysan** [comment]
> If I understand correctly from your test case, you are saving a Parse File with a full URL to the database. 

This is correct. Files are saved with url. And that url doesn't change after saving. File adapter is only used when we save object. And object keeps the first url they have got .




> A Parse File normally does not contain the absolute URL but only the filename. The file adapter only adds the base URL part when retrieving the file. 

This is how it should be. But in my database i can see the url

For example this is how it should be in database

`media: filename`

But i see this:
```
media: {
    __type:File,
    name: filename,
    url:fileUrl
}


```

**Comment by uzaysan** [comment]
>but you are not actually instantiating a new `Parse.File` according to the [docs](https://docs.parseplatform.org/js/guide/#files), e.g.
> 
> ```js
> const file = new Parse.File("myfile.txt", { base64: base64 });
> ```
> 

That's not true. See my example code in first message. İm creating a new file like this

`var mainmedia = new Parse.File(user.id, { base64: mediaData },mimeInfo.mime);`

But this shouldn't be the issue here

**Comment by noaker** [comment]
@uzaysan have you found the solution to this yet? I have experienced a similar problem

When I tag the created (and saved) ParseFile to my ParseObject and proceed to save the ParseObject, the ParseFile is saved as an Array type, rather than the File type.

Codes in Android Studio:
```
// Create and save file
ParseFile parseFile = new ParseFile("image.jpg", byteArray);
parseFile.saveInBackground();

// Tag file to ParseObject and save
ParseObject parseObject = new ParseObject("File");
parseObject.add("file", parseFile);
parseObject.add("user", ParseUser.getCurrentUser());
parseObject.saveInBackground();
```

The data type for "file" is File.

This is the error that I received: 
`schema mismatch for File.file; expected File but got Array`

After changing the data type to Array, this is the content of my "file" field in the database
```
[
  {
    "__type": "File",
    "name": "b8c7974254a3edbfe6b9075a63cf728b_image.jpg",
    "url": "https://linktomyimage"
  }
]
```

In addition to ParseFile, I also saved the user to the ParseObject. Strangely, the same happens to the data type for the user field - it becomes an array, rather than Pointer. Forcing it as a Pointer results in an error.

**Comment by uzaysan** [comment]
> When I tag the created (and saved) ParseFile to my ParseObject and proceed to save the ParseObject, the ParseFile is saved as an Array type, rather than the File type.

Your problem is different than mine. But its strange. I was also using android sdk and didnt face this issue. But I was passing android.File instead of byte array. What version of android sdk are you using? Can you try to pass file instead of byteArray?

Here is an example code:
```
// uri is the uri you get on activityResult call
File file = new File(uri.toString());
ParseFile parseFile = new ParseFile(file);
parseFile.saveInBackground(new SaveCallback() {
    @Override
    public void done(ParseException e) {
        ParseObject parseObject = new ParseObject("File");
        parseObject.add("file", parseFile);
        parseObject.add("user", ParseUser.getCurrentUser());
        parseObject.saveInBackground();                                                    
    }
});

//........


```
